app/controllers/items/detail.js

```diff
@@ -2,204 +2,345 @@ import Ember from "ember";
-  ),
-
-  showPieces: Ember.computed.alias("model.code.allow_pieces"),
-
-  tabName: Ember.computed("currentRoute", function() {
-    return this.get("currentRoute")
-      .split(".")
-      .get("lastObject");
-  }),
-
-  grades: Ember.computed("item.grade", function() {
-    return [
-      { name: "A", id: "A" },
-      { name: "B", id: "B" },
-      { name: "C", id: "C" },
-      { name: "D", id: "D" }
-    ];
-  }),
-
-  conditions: Ember.computed(function() {
-    return this.get("store").peekAll("donor_condition");
-  }),
-
-  itemMarkedMissing: Ember.observer("item.inventoryNumber", function() {
-    if (
-      !this.get("item.inventoryNumber") &&
-      this.get("target").currentPath === "items.detail"
-    ) {
-      this.get("messageBox").alert(
-        "This item is not inventoried yet or has been marked as missing.",
-        () => {
-          this.transitionToRoute("items.index");
+    tabName: Ember.computed("currentRoute", function() {
+      return this.get("currentRoute")
+        .split(".")
+        .get("lastObject");
+    }),
+
+    conditions: Ember.computed(function() {
+      return this.get("store").peekAll("donor_condition");
+    }),
+
+    isItemDetailPresent() {
+      return !!this.get("item.detail.length");
+    },
+
+    displayFields: Ember.computed("model.code", function() {
+      let subform = this.get("model.code.subform");
+      return this.returnDisplayFields(subform);
+    }),
+
+    selectedCountry: Ember.computed("item.detail", function() {
+      let country = this.get("item.detail.country");
+      if (country) {
+        return {
+          id: country.id,
+          nameEn: country.get("nameEn")
+        };
+      }
+    }),
+
+    returnSelectedValues(selectedValues) {
+      let dataObj = {
+        ...this.get("dataObjnew")
+      };
+      Object.keys(selectedValues).map((data, index) => {
+        if (this.get("fixedDropdownArr").indexOf(data) >= 0) {
+          let subformColumn = `${data.substring(0, data.length - 2)}`;
+          dataObj[subformColumn] = {
+            id: this.get(`item.detail.${data}`),
+            tag: this.get(`item.detail.${subformColumn}.labelEn`)
+          };
+        } else {
+          dataObj[data] = {
+            id: index + 1,
+            tag: selectedValues[data]
+          };
         }
-      );
-    }
-  }),
+      });
+      return { ...dataObj };
+    },
 
-  performDispatch: Ember.observer("showDispatchOverlay", function() {
-    Ember.run.debounce(this, this.updateAutoDisplayOverlay, 100);
-  }),
+    selectedValuesDisplay: Ember.computed("item.detail", "dataObjnew", {
+      get(key) {
+        if (!this.get("showAdditionalFields")) {
+          return false;
+        }
+        let selectedValues = this.get("item.detail.data");
+        let returnData = this.returnSelectedValues(selectedValues);
+        this.set("dataObjnew", {
+          ...returnData
+        });
+        return {
+          ...this.get("dataObjnew")
+        };
+      },
```

Comment:
I'm assuming you're using the spread operator to create a "clone" and to avoid changes to propagate to `dataObjnew` (very vague name btw, might be worth changing)

Since the `returnData`  ref is not used after being copied into `dataObjnew`, I don't know if the second use of the spread operator is needed. You might as well return it directly : )